packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -111,27 +136,55 @@ export default function MongoNav({
   urls: urlsProp,
   constructOrganizationURL: constructOrganizationURLProp,
   constructProjectURL: constructProjectURLProp,
+  onError = () => {},
+  onSuccess = () => {},
 }: MongoNavInterface) {
-  let data;
+  const [data, setData] = React.useState<DataInterface | undefined>(undefined);
 
-  if (mode === Mode.Dev) {
-    data = devModeData;
+  const hosts = defaultsDeep(hostsProp, defaultHosts);
+  const endpointURI = `${hosts.cloud}/user/shared`;
+
+  function getProductionData() {
+    return fetch(endpointURI, {
+      credentials: 'include',
+      method: 'GET',
+    });
   }
 
-  if (!data) {
-    // Eventually this logic will be more robust, but for an alpha version will return null without data
-    return null;
+  function getFixtureData() {
+    return new Promise(resolve => {
+      onSuccess?.(fixtureData);
+
+      resolve(fixtureData);
+    });
   }
 
-  const {
-    account,
-    currentOrganization,
-    currentProject,
-    organizations,
-    projects,
-  } = data;
+  async function handleResponse(response: Response) {
+    if (!response.ok) {
+      const status = response.status as 401; //typecasting for now until we have more types to handle
+      onError?.(ErrorCodeMap[status]);
+      console.error(ErrorCodeMap[status]);
+    } else {
+      const data = await response.json();
+      setData(data);
+      onSuccess?.(data);
```

Comment:
Let's be consistent with the order in which we call the callback and the internal functions. Is it prop callback first, then internal functions, or vice versa? The order is swapped in getFixtureData.

Reply:
👍